backend/api/projects.py

```diff
@@ -149,8 +160,16 @@ def get_project_by_id(project_id):
     Returns:
         res (json): Project corresponding to the ID
     """
-    res = g.projects.find_one({'_id': project_id})
-    if not res:
+    archived = request.args.get('archived', type=str, default="false")
+    if archived == 'true':
+        res = g.projects.find_one({'_id': project_id, 'archived': True})
+    elif archived == 'false':
+        res = g.projects.find_one({'_id': project_id, 'archived': False})
+    elif archived == 'mixed':
+        res = g.projects.find_one({'_id': project_id})
+    else:
+        res = None
```

Comment:
I don't this getting a project directly via Id should require archived to be set, now that we allow everybody to search in all archived projects. Could you revert this part?